--- configure.orig	2004-02-23 15:03:04.000000000 -0500
+++ configure	2004-02-23 15:06:54.000000000 -0500
@@ -7583,7 +7583,7 @@
   ssllib="-L/usr/lib -lssl -lcrypto"
 fi
 if test -d '/usr/local/ssl/include'; then
-  sslinc="-I/usr/local/ssl/include"
+  sslinc="-I/usr/local/ssl/include/openssl -I/usr/local/ssl/include"
 elif test -d '/usr/local/openssl/include'; then
   sslinc="-I/usr/local/openssl/include"
 elif test -d '/usr/local/include/openssl'; then
@@ -7611,8 +7611,8 @@
   *)
     if test "x$withval" = "xyes"; then
       withval=$ssllib
-      SSLINC=$sslinc
     fi
+    SSLINC=$sslinc
     LIBS="$LIBS $withval"
     cat > conftest.$ac_ext <<EOF
 #line 7619 "configure"
