--- php-seclib-0.2.1/Crypt/TripleDES.php~	2010-02-25 22:40:26.000000000 +0200
+++ php-seclib-0.2.1/Crypt/TripleDES.php	2010-08-31 09:54:45.919440789 +0300
@@ -54,7 +54,7 @@
 /**
  * Include Crypt_DES
  */
-require_once 'DES.php';
+require_once 'Crypt/DES.php';
 
 /**
  * Encrypt / decrypt using inner chaining
--- php-seclib-0.2.1/Crypt/AES.php~	2010-02-09 01:10:26.000000000 +0200
+++ php-seclib-0.2.1/Crypt/AES.php	2010-08-31 09:56:04.703391530 +0300
@@ -63,7 +63,7 @@
 /**
  * Include Crypt_Rijndael
  */
-require_once 'Rijndael.php';
+require_once 'Crypt/Rijndael.php';
 
 /**#@+
  * @access public
