diff -ur mongodb-src-r2.2.0.orig/src/SConscript.client mongodb-src-r2.2.0/src/SConscript.client
--- mongodb-src-r2.2.0.orig/src/SConscript.client	2012-08-28 00:28:11.000000000 -0500
+++ mongodb-src-r2.2.0/src/SConscript.client	2012-10-02 11:20:48.968121332 -0500
@@ -98,7 +98,7 @@
 mongoclient_lib = env.Library('mongoclient', clientSource),
 mongoclient_install = env.Install('#/', [
         mongoclient_lib,
-        #env.SharedLibrary('mongoclient', clientSource),
+        env.SharedLibrary('mongoclient', clientSource),
         ])
 env.Alias('mongoclient', mongoclient_install)
 
