diff --git a/apps/snmptrapd_sql.c b/apps/snmptrapd_sql.c
index ccba258..56832a1 100644
--- a/apps/snmptrapd_sql.c
+++ b/apps/snmptrapd_sql.c
@@ -437,11 +437,7 @@ netsnmp_mysql_init(void)
         return -1;
     }
 
-#ifdef HAVE_BROKEN_LIBMYSQLCLIENT
     my_init();
-#else
-    MY_INIT("snmptrapd");
-#endif
 
     /** load .my.cnf values */
     load_defaults ("my", _sql.groups, &not_argc, &not_argv);
