Index: lib/rubygems/commands/install_command.rb
===================================================================
--- lib/rubygems/commands/install_command.rb.orig	2008-09-10 19:54:41.000000000 -0400
+++ lib/rubygems/commands/install_command.rb	2008-10-21 12:17:11.000000000 -0400
@@ -72,6 +72,7 @@
       :format_executable => options[:format_executable],
       :ignore_dependencies => options[:ignore_dependencies],
       :install_dir => options[:install_dir],
+      :build_root => options[:build_root],
       :security_policy => options[:security_policy],
       :wrappers => options[:wrappers],
       :bin_dir => options[:bin_dir],
Index: lib/rubygems/lib/rubygems/dependency_installer.rb
===================================================================
--- lib/rubygems/dependency_installer.rb.orig	2008-08-12 17:50:22.000000000 -0400
+++ lib/rubygems/dependency_installer.rb	2008-10-21 12:25:20.000000000 -0400
@@ -58,6 +58,7 @@
     @force = options[:force]
     @format_executable = options[:format_executable]
     @ignore_dependencies = options[:ignore_dependencies]
+    @build_root = options[:build_root]
     @security_policy = options[:security_policy]
     @user_install = options[:user_install]
     @wrappers = options[:wrappers]
@@ -241,6 +242,7 @@
                                 :format_executable   => @format_executable,
                                 :ignore_dependencies => @ignore_dependencies,
                                 :install_dir         => @install_dir,
+                                :build_root          => @build_root,
                                 :security_policy     => @security_policy,
                                 :source_index        => @source_index,
                                 :user_install        => @user_install,
Index: lib/rubygems/lib/rubygems/installer.rb
===================================================================
--- lib/rubygems/installer.rb.orig	2008-09-23 00:01:38.000000000 -0400
+++ lib/rubygems/installer.rb	2008-10-21 12:17:11.000000000 -0400
@@ -106,7 +106,13 @@
     @format_executable = options[:format_executable]
     @security_policy = options[:security_policy]
     @wrappers = options[:wrappers]
-    @bin_dir = options[:bin_dir]
+    @bin_dir = options[:bin_dir] || Gem.bindir(@gem_home)
+    @build_root = options[:build_root]
+    unless @build_root.nil? or @build_root == ""
+        @build_root = Pathname.new(@build_root).expand_path
+        @gem_home = File.join(@build_root,@gem_home)
+        @bin_dir = File.join(@build_root,@bin_dir)
+    end
     @development = options[:development]
     @source_index = options[:source_index]
 
Index: lib/rubygems/lib/rubygems/install_update_options.rb
===================================================================
--- lib/rubygems/install_update_options.rb.orig	2008-06-24 14:56:30.000000000 -0400
+++ lib/rubygems/install_update_options.rb	2008-10-21 12:17:11.000000000 -0400
@@ -31,6 +31,12 @@
       options[:bin_dir] = File.expand_path(value)
     end
 
+    add_option(:"Install", '-B', '--build-root DIR',
+      'Temporary installation root. Useful for building',
+      'packages. Do not use this when installing remote gems.') do |value, options|
+      options[:build_root] = File.expand_path(value)
+    end
+
     add_option(:"Install/Update", '-d', '--[no-]rdoc',
                'Generate RDoc documentation for the gem on',
                'install') do |value, options|
